src/mpyl/steps/steps.py

```diff
@@ -125,19 +125,22 @@ def _execute_stage(self, stage: Stage, project: Project, dry_run: bool = False)
                 artifact: Optional[Artifact] = self._find_required_artifact(project, executor.required_artifact)
                 result = Output(success=True, message='')
                 if executor.before:
-                    result = self._execute(executor.before, project, self._properties,
-                                           self._find_required_artifact(project, executor.before.required_artifact),
-                                           dry_run)
+                    for before_step in executor.before:
+                        result = self._execute(before_step, project, self._properties,
+                                               self._find_required_artifact(project, before_step.required_artifact),
+                                               dry_run)
```

Comment:
We should probably combine the results here then. Right now it just takes the last execution